AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -653,6 +698,15 @@ - (void)displayImmediately
   [[self asyncLayer] displayImmediately];
 }
 
+- (void)recursivelyDisplayImmediately
```

Comment:
Check out -recursivelyEnsureDisplay; will it do what you need?  If so (I am not confident with this quick review), it would be notably faster.